paddle/fluid/operators/slice_op.cc

```diff
@@ -155,7 +155,9 @@ class SliceOp : public framework::OperatorWithKernel {
 #ifdef PADDLE_WITH_MKLDNN
       auto input_data_type =
           framework::OperatorWithKernel::IndicateVarDataType(ctx, "Input");
-
+      auto vec_dims = phi::vectorize(in_tensor.dims());
+      bool not_all_zero_dim = std::all_of(
+          vec_dims.cbegin(), vec_dims.cend(), [](int64_t i) { return i != 0; });
       if (this->CanMKLDNNBeUsed(ctx, input_data_type)) {
```

Comment:
```suggestion
      if (!not_all_zero_dims && this->CanMKLDNNBeUsed(ctx, input_data_type)) {
```
This will result in performing less unnecessary operations when not_all_zero_dims is true.

Reply:
Done